pageserver/src/tenant/mgr.rs

```diff
@@ -261,6 +261,10 @@ pub struct TenantManager {
     // See https://github.com/neondatabase/neon/issues/5796
     tenants: &'static std::sync::RwLock<TenantsMap>,
     resources: TenantSharedResources,
+
+    // This token is for use in edge cases like tenant deletion, where we are doing work on a
+    // tenant after it has been shut down, during deletion.
+    cancel: CancellationToken,
 }
```

Comment:
The description sounds tenant-scoped, why is it in TenantManager?

It'll probably evident when reading the rest of the PR, but, on first sight, it feels like this is missing some context.


```suggestion

    /// Long-running activities that operate on the tenant namespace instead of "inside" a Tenant
    /// should be sensitive to this token.
    ///
    /// Cancelled on [`TenantManager::shutdown`]
    cancel: CancellationToken,
}
```

Reply:
Yes, this comment wasn't very well worded.

Rewritten to:
```
    // Long-running operations that happen outside of a [`Tenant`] lifetime should respect this token.
    // This is for edge cases like tenant deletion.  In normal cases (within a Tenant lifetime),
    // tenants have their own cancellation tokens, which we fire individually in [`Self::shutdown`], or
    // when the tenant detaches.
```